tests/test-ImageSizes.php

```diff
@@ -0,0 +1,581 @@
+	 * @dataProvider get_size_data_for_standardise_size_data
+	 */
+	public function test__standardise_size_data( $size_data, $expected ) {
+		$attachment_id = self::factory()->attachment->create_object(
+			$this->test_image, 0, [
+				'post_mime_type' => 'image/jpeg',
+				'post_type'      => 'attachment',
+			]
+		);
+
+		wp_update_attachment_metadata( $attachment_id, wp_generate_attachment_metadata( $attachment_id, $this->test_image ) );
+
+		$postmeta = get_post_meta( $attachment_id, '_wp_attachment_metadata', true );
+
+		$imageSizes = new A8C_Files\ImageSizes( $attachment_id, $postmeta );
+
+		$standardised_size_data = $imageSizes->standardise_size_data( $size_data );
+
+		$this->assertEquals( $expected, $standardised_size_data );
+	}
+
+	/**
+	 * Default image sizes as defined in Database schema.
+	 *
+	 * @return array Default image sizes.
+	 */
+	public function default_sizes() {
+		return [
+			'thumbnail' => [
+				'width' => 150,
+				'height' => 150,
+				'calculated_dimensions' => [
+					'width' => 150,
+					'height' => 150,
+				],
+				'params' => [
+					'resize' => '150,150',
+				],
+			],
+			'medium' => [
+				'width' => 300,
+				'height' => 300,
+				'calculated_dimensions' => [
+					'width' => 300,
+					'height' => 169,
+				],
+				'params' => [
+					'resize' => '300,169',
+				],
+			],
+			'medium_large' => [
+				'width' => 768,
+				'height' => 0,
+				'calculated_dimensions' => [
+					'width' => 768,
+					'height' => 432,
+				],
+				'params' => [
+					'resize' => '768,432',
+				],
+			],
+			'large' => [
+				'width' => 1024,
+				'height' => 1024,
+				'calculated_dimensions' => [
+					'width' => 1024,
+					'height' => 576,
+				],
+				'params' => [
+					'resize' => '1024,576',
+				],
+			],
+		];
+	}
+
+	/**
+	 * Integration test of the original behaviour of the VIP Go platform:
+	 *
+	 * 1. no intermediate image sizes are being physically created.
+	 * 2. no intermediate image sizes are being filtered in.
+	 */
+	public function test__original_behaviour_on_vip_go() {
+
+		$this->disable_image_sizes();
+
+		$attachment_id = self::factory()->attachment->create_object(
+			$this->test_image, 0, [
+				'post_mime_type' => 'image/jpeg',
+				'post_type'      => 'attachment',
+			]
+		);
+
+		wp_update_attachment_metadata( $attachment_id, wp_generate_attachment_metadata( $attachment_id, $this->test_image ) );
+
+		$postmeta = get_post_meta( $attachment_id, '_wp_attachment_metadata', true );
+
+		// This means no intermediate image sizes were physically created.
+		$this->assertEmpty( $postmeta['sizes'] );
+
+		$metadata = wp_get_attachment_metadata( $attachment_id );
+
+		// This means the sizes are not being created via any filters.
+		$this->assertEmpty( $metadata['sizes'] );
+	}
+
+	/**
+	 * Integration test of the virtual creation of intermediate sizes.
+	 * No physical copies are being created.
+	 */
+	public function test__inject_image_sizes() {
```

Comment:
It would be good to have a test for a non-image mime type as well.

Reply:
Very good call. Addressed via https://github.com/Automattic/vip-go-mu-plugins/pull/860/commits/df55f2db5f971cb6232ac7b1b29d08c9c379deee